utils/training.py

```diff
@@ -64,3 +93,36 @@ def on_epoch_end(self, epoch, logs=None):
     logs.update({"lr": K.eval(self.model.optimizer.lr)})
     super().on_epoch_end(epoch, logs)
 
+class VisualizeCallback(tf.keras.callbacks.Callback):
+  """
+  Mixture weights visualization callback.
+
+  Arguments:
+  save_path: the mixture weight plots will be saved in this directory.
+    
+  """
+  def __init__(self, save_path, domains, num_templates, num_layers,
+               frequency=10, **args):
+    self.frequency = frequency
+    self.save_path = os.path.abspath(save_path)
+    self.domains = domains
+    self.num_templates = num_templates
+    self.num_layers = num_layers
+    # self.summary = tf.summary.create_file_writer(os.path.join(self.save_path,
+    #                                                           "imglog"))
+    if not os.path.exists(self.save_path):
+      os.makedirs(self.save_path)
+    super(VisualizeCallback, self).__init__(**args)
+
+
+  def on_epoch_end(self, epoch, logs=None):
+    if epoch % self.frequency == 0:
+      mw_img = visualize_mixture_weights(self.model, domains=self.domains,
+                                         num_templates=self.num_templates,
+                                         num_layers=self.num_layers) 
+      fname = os.path.join(self.save_path, "mixtures_%d.png" % epoch) 
+      imageio.imwrite(fname, mw_img)
+      # with self.summary.as_default():
+      #   tf.summary.image("Mixture weights", mw_img, step=epoch)
+      #   tf.summary.scalar("A", 1., step=epoch)
```

Comment:
Instead of leaving as a comment, this could be turned into a working code controlled by a flag or a global boolean flag (enabling or disabling this behavior).